connectionpool.go

```diff
@@ -150,10 +145,10 @@ func (p *policyConnPool) SetHosts(hosts []*HostInfo) {
 			// don't create a connection pool for a down host
 			continue
 		}
-		ip := host.ConnectAddress().String()
-		if _, exists := p.hostConnPools[ip]; exists {
+		hostID := MustParseUUID(host.HostID())
```

Comment:
Good point, I used MustParse because I assumed that if `UUID` type is returned by the driver, this means the typed value is correct UUID, and there's only single place where this field is initialized from typed `UUID`
https://github.com/scylladb/gocql/blob/b4695642d1cb72da10706ce51b99c94b29e1653a/host_source.go#L455-L460

I digged down and unfortunately driver just copies what it's returned from the server-side, so invalid uuid may be returned.